larray_editor/arraywidget.py

```diff
@@ -537,35 +530,37 @@ def __init__(self, parent, data=None, readonly=False, bg_value=None, bg_gradient
             readonly = True
         self.readonly = readonly
 
+        # prepare internal views and models
         self.model_axes = LabelsArrayModel(parent=self, readonly=readonly)
         self.view_axes = LabelsView(parent=self, model=self.model_axes, position=(TOP, LEFT))
 
-        self.model_xlabels = LabelsArrayModel(parent=self, readonly=readonly)
-        self.view_xlabels = LabelsView(parent=self, model=self.model_xlabels, position=(TOP, RIGHT))
+        self.model_hlabels = LabelsArrayModel(parent=self, readonly=readonly)
+        self.view_hlabels = LabelsView(parent=self, model=self.model_hlabels, position=(TOP, RIGHT))
 
-        self.model_ylabels = LabelsArrayModel(parent=self, readonly=readonly)
-        self.view_ylabels = LabelsView(parent=self, model=self.model_ylabels, position=(BOTTOM, LEFT))
+        self.model_vlabels = LabelsArrayModel(parent=self, readonly=readonly)
+        self.view_vlabels = LabelsView(parent=self, model=self.model_vlabels, position=(BOTTOM, LEFT))
 
         self.model_data = DataArrayModel(parent=self, readonly=readonly, minvalue=minvalue, maxvalue=maxvalue)
         self.view_data = DataView(parent=self, model=self.model_data)
+        self.model_data.dataChanged.connect(parent.data_changed)
```

Comment:
I don't like this change because it makes an assumption on how the widget will be used. In other words, it makes the coupling between widget and its container stronger instead of looser, the later being usually better. The widget can be used in several ways, so assuming the parent has a data_changed method/signal isn't a good idea. What we probably want to do instead is to (re)emit the data_changed signal at the level of the widget so that the parent does not need to know about the model. The parent should still be responsible to connect to that signal though.